hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -169,75 +190,93 @@ public void handleRound(@NonNull final HederaState state, @NonNull final Round r
+        Fees fees = null;
         try {
-            // Setup configuration
-            var configuration = configProvider.getConfiguration();
-            final var hederaConfig = configuration.getConfigData(HederaConfig.class);
+            final var preHandleResult =
+                    getCurrentPreHandleResult(state, platformEvent, creator, platformTxn, configuration);
 
-            preHandleResult = getCurrentPreHandleResult(state, platformEvent, platformTxn, configuration);
             final var transactionInfo = preHandleResult.txInfo();
+
+            if (transactionInfo == null) {
+                // FUTURE: Charge node generic penalty, set values in record builder, and remove log statement
+                logger.error("Non-parsable transaction from creator {}", creator);
+                return null;
+            }
+
+            // Get the parsed data
             final var transaction = transactionInfo.transaction();
-            final var txBody = transactionInfo.txBody();
-            final Bytes transactionBytes;
+            txBody = transactionInfo.txBody();
+            payer = preHandleResult.payer();
 
+            final Bytes transactionBytes;
             if (transaction.signedTransactionBytes().length() > 0) {
                 transactionBytes = transaction.signedTransactionBytes();
             } else {
                 // in this case, recorder hash the transaction itself, not its bodyBytes.
                 transactionBytes = Bytes.wrap(PbjConverter.fromPbj(transaction).toByteArray());
             }
 
+            // Initialize record builder list
             recordBuilder
                     .transaction(transactionInfo.transaction())
                     .transactionBytes(transactionBytes)
                     .transactionID(txBody.transactionID())
                     .exchangeRate(exchangeRateManager.exchangeRates())
                     .memo(txBody.memo());
 
-            // If pre-handle was successful, we return the result. Otherwise, we charge the node or throw an exception.
-            switch (preHandleResult.status()) {
-                case SO_FAR_SO_GOOD -> {
-                    /* All good nothing to do */
-                }
-                case NODE_DUE_DILIGENCE_FAILURE -> createPenaltyPayment();
-                case UNKNOWN_FAILURE -> throw new IllegalStateException("Pre-handle failed with unknown failure");
-                default -> throw new PreCheckException(preHandleResult.responseCode());
-            }
-
-            // Check the time box of the transaction
-            checker.checkTimeBox(txBody, consensusNow);
-
-            // Check all signature verifications. This will also wait, if validation is still ongoing.
+            // Set up the verifier
+            final var hederaConfig = configuration.getConfigData(HederaConfig.class);
             final var verifier = new BaseHandleContextVerifier(hederaConfig, preHandleResult.verificationResults());
```

Comment:
Should this line be moved down under `dispatcher.dispatchComputeFees`? It doesn't look like it needs to come before. Does it throw an exception that requires it to come before?

Reply:
NOPE: It is needed in the context